src/components/segmentedControl/segmentedControl.svelte

```diff
@@ -39,10 +39,16 @@
 
       if (value === getValue(controlItem)) {
         controlItem.setAttribute('aria-selected', '')
-        window.requestAnimationFrame(() => {
+        const zero = document.timeline.currentTime as number;
+        function setPill(t: number) {
+          const timeDiff = t - zero;
           pillWidth = controlItem.getBoundingClientRect().width
           pillPosition = controlItem.offsetLeft
-        })
+          if (timeDiff < 1000) {
+            window.requestAnimationFrame(setPill)
+          }
+        }
+        window.requestAnimationFrame(setPill)
```

Comment:
This may not be a great way to do this, but I'm running into a problem where the initial width isn't _quite_ large enough, so the theory here is to repeatedly request an animation for reasonably long enough period to assume the final width is correct.